src/sources/vector/v2.rs

```diff
@@ -0,0 +1,156 @@
+use crate::{
+    config::SourceContext,
+    config::{DataType, GenerateConfig, Resource, SourceConfig},
+    event::Event,
+    proto::vector as proto,
+    shutdown::{ShutdownSignal, ShutdownSignalToken},
+    sources::Source,
+    Pipeline,
+};
+
+use futures::{FutureExt, SinkExt, TryFutureExt};
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use tonic::{
+    transport::{Certificate, Identity, Server, ServerTlsConfig},
+    Request, Response, Status,
+};
+
+#[derive(Debug, Clone)]
+pub struct Service {
+    pipeline: Pipeline,
+}
+
+#[tonic::async_trait]
+impl proto::Service for Service {
+    async fn push_events(
+        &self,
+        request: Request<proto::EventRequest>,
+    ) -> Result<Response<proto::EventAck>, Status> {
+        let event = request
+            .into_inner()
+            .message
+            .map(Event::from)
+            .ok_or(Status::invalid_argument("missing event"))?;
+
+        let response = Response::new(proto::EventAck {
+            // TODO: There is no need for any body in the ack.
+            message: "success".to_owned(),
+        });
+
+        self.pipeline
+            .clone()
+            .send(event)
+            .await
+            .map(|_| response)
+            .map_err(|err| Status::unavailable(err.to_string()))
+    }
+
+    // TODO: figure out a way to determine if the current Vector instance is "healthy".
```

Comment:
This was discussed via chat. We're going to always report "healthy" for now, and worry about getting actual pipeline health status embedded at a later stage.